internal/compute/provider.go

```diff
@@ -55,6 +44,62 @@ func NewComputeProvider(provider models.ProviderID) (Provider, error) {
 }
 
 // NewProvisioner creates a new system provisioner
-func NewProvisioner(jobID string) Provisioner {
-	return NewAnsibleConfigurator(jobID)
+func NewProvisioner(jobID string) provisioner.Provisioner {
+	// Extract job name from job ID (format: "job-20250411-134559")
+	jobName := strings.TrimPrefix(jobID, "job-")
+
+	cfg := &config.AnsibleConfig{
+		JobID:             jobID,
+		JobName:           jobName,
+		SSHUser:           provisioner.DefaultSSHUser,
+		SSHKeyPath:        os.ExpandEnv(provisioner.DefaultSSHKeyPath),
+		PlaybookPath:      provisioner.PathToPlaybook,
+		InventoryBasePath: provisioner.InventoryBasePath,
+	}
+
+	p, err := provisioner.NewAnsibleProvisioner(cfg)
+	if err != nil {
+		// This should never happen with our default config
+		panic(fmt.Sprintf("failed to create provisioner: %v", err))
+	}
+	return p
+}
+
+// validateJobID validates the format of a job ID (should be "job-YYYYMMDD-HHMMSS")
+func validateJobID(jobID string) error {
+	if jobID == "" {
+		return fmt.Errorf("job ID is required")
+	}
+
+	if !strings.HasPrefix(jobID, "job-") {
+		return fmt.Errorf("invalid job ID format: must start with 'job-'")
+	}
+
+	parts := strings.Split(jobID, "-")
+	if len(parts) != 3 {
+		return fmt.Errorf("invalid job ID format: must be 'job-YYYYMMDD-HHMMSS'")
+	}
+
+	return nil
+}
+
+// NewAnsibleProvisioner creates a new Ansible provisioner
+func NewAnsibleProvisioner(jobID string) (provisioner.Provisioner, error) {
```

Comment:
what is going on here? why is there a compute `NewAnsibleProvisioner` and a `provisioner.NewAnsibleProvisioner`?